kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/test/delayController.kt

```diff
@@ -1,18 +1,12 @@
 package io.kotest.core.test
 
-import kotlinx.coroutines.CoroutineDispatcher
-import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.test.TestCoroutineScheduler
-import kotlinx.coroutines.test.TestDispatcher
 import kotlin.coroutines.CoroutineContext
 
-@ExperimentalStdlibApi
 val TestScope.testCoroutineScheduler: TestCoroutineScheduler
    get() = coroutineContext.testCoroutineScheduler
 
-@ExperimentalStdlibApi
 val CoroutineContext.testCoroutineScheduler: TestCoroutineScheduler
-   get() = when (val dispatcher = this[CoroutineDispatcher]) {
-      is TestDispatcher -> dispatcher.scheduler
-      else -> error("CoroutineDispatcher is not a TestDispatcher [$dispatcher]")
+   get() = checkNotNull(this[TestCoroutineScheduler]) {
+      "TestCoroutineScheduler not found. Make sure the option 'coroutineTestScope' is enabled."
```

Comment:
I'm not sure the error description is correct, so any suggestions on it are appreciated